ArcGISRuntimeSDKQt_QMLSamples/Routing/OfflineRouting/OfflineRouting.qml

```diff
@@ -0,0 +1,234 @@
+
+            onComponentCompleted: {
+                load();
+            }
+
+            onLoadStatusChanged: {
+                if (loadStatus === Enums.LoadStatusLoaded) {
+                    // assign ComboBox model from travel mode names
+                    let modesList = routeTask.routeTaskInfo.travelModes;
+                    let travelModeNames = [];
+                    for (let i = 0; i < modesList.length; i++) {
+                        travelModeNames.push(modesList[i].name);
+                    }
+                    comboBox.model = travelModeNames;
+                    createDefaultParameters();
+                }
+            }
+
+            function findRoute() {
+                if (routeTask.solveRouteStatus === Enums.TaskStatusInProgress)
+                    return;
+
+                if (createDefaultParametersResult === null)
+                    return;
+
+                let routeParameters = createDefaultParametersResult;
+                let stopsList = [];
+                let numGraphics = stopsOverlay.graphics.count;
+                if (numGraphics > 1) {
+                    for (let i = 0; i < numGraphics; i++) {
+                        let tempStop = ArcGISRuntimeEnvironment.createObject("Stop", {geometry: stopsOverlay.graphics.get(i).geometry});
+                        stopsList.push(tempStop);
+                    }
+                    routeParameters.setStops(stopsList);
+                    routeParameters.travelMode = routeTask.routeTaskInfo.travelModes[comboBox.currentIndex];
+
+                    routeTask.solveRoute(routeParameters);
+                }
+            }
+
+            onErrorChanged: {
+                console.log(routeTask.error.message, routeTask.error.additionalMessage);
+            }
+
+            onSolveRouteStatusChanged: {
+                if (solveRouteStatus === Enums.TaskStatusCompleted)
+                {
+                    // clear old route
+                    routeOverlay.graphics.clear();
+                    if (routeTask.error) {
+                        console.warn(routeTask.error.message);
+                    }
+                    let routeGraphic = ArcGISRuntimeEnvironment.createObject("Graphic", {geometry: solveRouteResult.routes[0].routeGeometry});
+                    routeOverlay.graphics.append(routeGraphic);
+                }
+            }
+        }
+
+        onIdentifyGraphicsOverlayStatusChanged: {
+            if (identifyGraphicsOverlayStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            if (identifyGraphicsOverlayResult === null) {
+                return;
+            }
+
+            if (identifyGraphicsOverlayResult.graphics.length === 0) {
+                selectedGraphic = null;
+            } else {
+                for (let i = 0; i < stopsOverlay.graphics.count; i++) {
+                    if(identifyGraphicsOverlayResult.graphics[0].equals(stopsOverlay.graphics.get(i))) {
```

Comment:
bounds check

Reply:
Is this not addressed by 
```C++
if (identifyGraphicsOverlayResult.graphics.length === 0) {
                selectedGraphic = null;
            }
```
? Or do I need to explicitly check `stopsOverlay.graphics` as well? @lsmallwood

Reply:
nope that sounds fine